src/prayer/AnswerPrayerForm/AnswerPrayerFormConnected.js

```diff
@@ -19,33 +38,71 @@ class AnswerPrayerFormConnected extends React.Component {
             currentUser: { profile: { photo = { uri: '' } } = {} } = {},
           } = {},
         }) => (
-          <Mutation mutation={ANSWER_PRAYER}>
-            {(answerPrayer) => (
-              <AnswerPrayerForm
-                loading={profileLoading}
-                onSubmit={(values) => {
-                  answerPrayer({
-                    variables: {
-                      id: values.id,
-                      answer: values.answer,
-                    },
-                    refetchQueries: () => [
-                      { query: GET_PRAYER_FEED, variables: { type: 'USER' } },
-                    ],
-                  });
-                  this.props.navigation.pop();
-                }}
-                avatarSource={photo}
-                {...this.props}
-                onClose={() => this.props.navigation.pop()}
-                title={"Celebrate God's faithfulness"}
-                prayerId={this.props.navigation.getParam('prayerId', '')}
-                prayerText={this.props.navigation.getParam('prayerText', '')}
-                prayerAnswer={this.props.navigation.getParam(
-                  'prayerAnswer',
-                  ''
+          <Mutation mutation={INTERACT_WITH_PRAYER}>
+            {(interactWithPrayer) => (
+              <Mutation mutation={ANSWER_PRAYER}>
+                {(answerPrayer) => (
+                  <AnswerPrayerForm
+                    loading={profileLoading}
+                    onSubmit={(values) => {
+                      answerPrayer({
+                        variables: {
+                          id: values.id,
+                          answer: values.answer,
+                        },
+                        refetchQueries: () => [
+                          {
+                            query: GET_PRAYER_FEED,
+                            variables: { type: 'USER' },
+                          },
+                        ],
+                      });
+                      this.props.navigation.pop();
+                    }}
+                    avatarSource={photo}
+                    {...this.props}
+                    onClose={() => this.closePrayer()}
+                    title={"Celebrate God's faithfulness"}
+                    prayerId={this.prayerId}
+                    prayerText={this.props.navigation.getParam(
+                      'prayerText',
+                      ''
+                    )}
+                    prayerAnswer={this.prayerAnswer}
+                    action={
+                      this.prayerAnswer ? (
+                        <OptionMenu>
```

Comment:
is there a reason you wouldn't just include this in the action component?

Reply:
You're talking about `prayerAnswer`? If so, it's because I use `prayerAnswer` as part of the initial values on the `AnswerPrayerForm` for Formik.

Reply:
I still don't understand why you can't just pass all three pieces in as one prop. you get what you can from the nav, stuff that and anything else into a `prayer` object, and then pass that in as a prop.